Spatial_Engine/Query/IsFullyOnLevel.cs

```diff
@@ -0,0 +1,117 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.oM.Dimensional;
+using BH.oM.Geometry;
+using BH.oM.Quantities.Attributes;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Geometry.SettingOut;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.Engine.Base;
+using System;
+
+namespace BH.Engine.Spatial
+{
+    public static partial class Query
+    {
+        /******************************************/
+        /****  Public Methods                  ****/
+        /******************************************/
+
+        [Description("Checks if the geometrical representation of an IElement0D is fully within a set distance from a level elevation.")]
+        [Input("element0D", "The IElement0D that will be checked for proximity to the Level.")]
+        [Input("level", "The Level to use for evaulation.")]
+        [Input("maxDistance", "The maximum distance allowed from the Level for this method to return true.", typeof(Length))]
+        [Output("isFullyOnLevel", "A boolean which is true if the geometrical representation of the IElement0D is fully within a set distance from the level elevation.")]
+        public static bool IsFullyOnLevel(this IElement0D element0D, Level level, double maxDistance)
```

Comment:
I would default `maxDistance` to `BH.oM.Geometry.Tolerance.Distance` - to save people having to provide the maximum distance every time?

Reply:
not sure I would for this case. That tolerance is really really tight. methods with a maxDistance input in ML has not default values for this in general. This might be a different case though, but just a thought.

Reply:
I did some testing on the project I am working on, and using the standard Rhino tolerance it was consistently able to identify floor slabs on every level, as pulled straight from Revit - no model laundry snap to levels or anything like that. So since it seems to be good enough for at least this real world case, I've pushed a commit renaming `maxDistance` to `tolerance` and defaulting to the standard Tolerance as suggested. 

On an unrelated note, also reworked the IElement2D method - it no longer calls the IElement1D method, but is much simpler and I now wish I had done the same for IsNearLevel.